### PR TITLE
[DOC] Add missing default values to the docstrings in 'nilearn/glm' - part 2

### PR DESCRIPTION
**Pull request by anupriyakkumari**
Completes the following tasks from https://github.com/nilearn/nilearn/issues/3865:
 Added missing default values in the docstrings of the following files: 

## [nilearn/glm/first_level/first_level.py](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/first_level/first_level.py)
[nilearn/glm/first_level/first_level.py::first_level_from_bids - line: 1244](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/first_level/first_level.py#L1244)

## [nilearn/glm/first_level/design_matrix.py](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/first_level/design_matrix.py)
[nilearn/glm/first_level/design_matrix.py::make_first_level_design_matrix - line: 282](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/first_level/design_matrix.py#L282)

## [nilearn/glm/first_level/hemodynamic_models.py](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/first_level/hemodynamic_models.py)
[nilearn/glm/first_level/hemodynamic_models.py::spm_hrf - line: 87](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/first_level/hemodynamic_models.py#L87)
[nilearn/glm/first_level/hemodynamic_models.py::glover_hrf - line: 120](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/first_level/hemodynamic_models.py#L120)
[nilearn/glm/first_level/hemodynamic_models.py::spm_time_derivative - line: 200](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/first_level/hemodynamic_models.py#L200)
[nilearn/glm/first_level/hemodynamic_models.py::glover_time_derivative - line: 239](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/first_level/hemodynamic_models.py#L239)
[nilearn/glm/first_level/hemodynamic_models.py::spm_dispersion_derivative - line: 320](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/first_level/hemodynamic_models.py#L320)
[nilearn/glm/first_level/hemodynamic_models.py::glover_dispersion_derivative - line: 357](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/first_level/hemodynamic_models.py#L357)
[nilearn/glm/first_level/hemodynamic_models.py::compute_regressor - line: 701](https://github.com/nilearn/nilearn/blob/main/nilearn/glm/first_level/hemodynamic_models.py#L701)

### CODE REVIEW

File: CITATION.cff
Comment (Remi-Gau) [inline CITATION.cff:770]:
```suggestion
```

Given that this will be already done in #4656, let's not have it there.

File: doc/changes/latest.rst
Comment (Remi-Gau) [inline doc/changes/latest.rst:105]: Let's keep it for now but I think that we should be able a single line in the changelog that mentions all the PRs you did.

Something like:

```rst
- :bdg-primary:`Doc` Add missing default values to docstrings (:gh:`4656`, :gh:`4659` by `Anupriya Kumari`_).
```

File: nilearn/glm/first_level/design_matrix.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/design_matrix.py:333]: In this case could also update the doc string to also mention.

```rst
Will be set to ``[0]`` if ``None`` is passed.
```

File: nilearn/glm/first_level/first_level.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/first_level.py:314]: In this case could also update the doc string to also mention.

```rst
Will be set to ``[0]`` if ``None`` is passed.
```

File: nilearn/glm/first_level/design_matrix.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/design_matrix.py:302]:
```suggestion
    events : DataFrame instance or None, default=None
```

File: nilearn/glm/first_level/design_matrix.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/design_matrix.py:333]:
```suggestion
    fir_delays : array of shape(n_onsets), :obj:`list` or None, default=None
```

File: nilearn/glm/first_level/design_matrix.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/design_matrix.py:339]:
```suggestion
            pandas DataFrame or None, default=None
```

File: nilearn/glm/first_level/design_matrix.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/design_matrix.py:343]:
```suggestion
    add_reg_names : :obj:`list` of (n_add_reg,) :obj:`str`, or None, default=None
```

File: nilearn/glm/first_level/first_level.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/first_level.py:288]:
```suggestion
    t_r : :obj:`float` or None, default=None
```

File: nilearn/glm/first_level/first_level.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/first_level.py:314]:
```suggestion
    fir_delays : array of shape(n_onsets), :obj:`list` or None, default=None
```

File: nilearn/glm/first_level/first_level.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/first_level.py:338]:
```suggestion
    target_shape : 3-tuple of :obj:`int`, or None, default=None
```

File: nilearn/glm/first_level/first_level.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/first_level.py:349]:
```suggestion
    memory_level : :obj:`int` or None, default=None
```

File: nilearn/glm/first_level/first_level.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/first_level.py:1299]:
```suggestion
    space_label : :obj:`str` or None, default=None
```

File: nilearn/glm/first_level/hemodynamic_models.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/hemodynamic_models.py:104]:
```suggestion
    time_length : :obj:`float`, default=32.0
```

File: nilearn/glm/first_level/hemodynamic_models.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/hemodynamic_models.py:107]:
```suggestion
    onset : :obj:`float`, default=0.0
```

File: nilearn/glm/first_level/hemodynamic_models.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/hemodynamic_models.py:137]:
```suggestion
    time_length : :obj:`float`, default=32.0
```

File: nilearn/glm/first_level/hemodynamic_models.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/hemodynamic_models.py:140]:
```suggestion
    onset : :obj:`float`, default=0.0
```

File: nilearn/glm/first_level/hemodynamic_models.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/hemodynamic_models.py:217]:
```suggestion
    time_length : :obj:`float`, default=32.0
```

File: nilearn/glm/first_level/hemodynamic_models.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/hemodynamic_models.py:220]:
```suggestion
    onset : :obj:`float`, default=0.0
```

File: nilearn/glm/first_level/hemodynamic_models.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/hemodynamic_models.py:256]:
```suggestion
    time_length : :obj:`float`, default=32.0
```

File: nilearn/glm/first_level/hemodynamic_models.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/hemodynamic_models.py:259]:
```suggestion
    onset : :obj:`float`, default=0.0
```

File: nilearn/glm/first_level/hemodynamic_models.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/hemodynamic_models.py:339]:
```suggestion
    time_length : :obj:`float`, default=32.0
```

File: nilearn/glm/first_level/hemodynamic_models.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/hemodynamic_models.py:342]:
```suggestion
    onset : :obj:`float`, default=0.0
```

File: nilearn/glm/first_level/hemodynamic_models.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/hemodynamic_models.py:376]:
```suggestion
    time_length : :obj:`float`, default=32.0
```

File: nilearn/glm/first_level/hemodynamic_models.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/hemodynamic_models.py:379]:
```suggestion
    onset : :obj:`float`, default=0.0
```

File: nilearn/glm/first_level/hemodynamic_models.py
Comment (Remi-Gau) [inline nilearn/glm/first_level/hemodynamic_models.py:727]:
```suggestion
    fir_delays : [int] 1D-array-like or None, default=None
```

In this case could also update the doc string to also mention.

```rst
Will be set to ``[0]`` if ``None`` is passed.
```

**Review by Remi-Gau — CHANGES_REQUESTED** [review]
Realized I forgot to mention that we should add `or None`, to the list of accepted types if we do not mention it.

Also if you have got the patience you can edit the type on the lines you modify from (for example)

```rst
int
```

to 

```rst
:obj:`int`
```

this allows automatic cross linking to the definition of that type in the python documentation:

see for example

image

**Review by Remi-Gau — APPROVED** [review]

**Comment by Remi-Gau** [comment]
- note to myself: merge #4656 before this one

**Comment by Remi-Gau** [comment]
> I have gone over all the files I edited in this PR and added the type wherever it was missing,

Thanks a bunch for doing that.

If you are interested I think that it may be able to modify to script we used to flag all the missing argument to even maybe automatically update the type in the doc strings from things like 

```
int
``` 

to 

```
:obj:`ìnt`
```

could save you / me / everyone a lot of time rather than doing those changes manually.

**Comment by anupriyakkumari** [comment]
Completely agreed. It's a hassle doing these changes manually. I will try to take a look at the script soon. Thanks!

**Comment by Remi-Gau** [comment]
that's the script

https://github.com/nilearn/nilearn_sandbox/blob/master/nilearn_sandbox/_utils/missing_docstring_default.py

my idea would be to write another script rather than modify this one

if you are interested let me know and I will open a separate issue for this

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Comment by Remi-Gau** [comment]
fixed the conflict in the change log

waiting for the doc build to finish in CI before merging

**Comment by Remi-Gau** [comment]
that was a mis-click did not mean to close

**Review by man-shu — APPROVED** [review]

**Comment by anupriyakkumari** [comment]
Hi @ Remi-Gau, I have been looking at the script, and I can take up the task of writing a script to change the types of parameters automatically. Could you open a new issue for that? Also, I would need some specifications, such as what kind of functions we need to change and whether we add Python objects or classes. Any additional resource that you think I should look at would also be great! Thanks

**Pull request merged**
